test/gtest/common/test_helpers.cc

```diff
@@ -127,3 +127,58 @@ message_stream::~message_stream() {
 } // detail
 
 } // ucs
+
+namespace ucp {
+
+data_type_desc_t &
+data_type_desc_t::make(ucp_datatype_t datatype, void *buf, size_t length,
+                       size_t iov_cnt)
+{
+    EXPECT_FALSE(is_valid());
+
+    if (_length == 0) {
+        _length = length;
+    }
+
+    if (_origin == uintptr_t(NULL)) {
```

Comment:
you compare with uintptr_t(NULL), but you initialize _origin by 0 in constructor. is this correct?